lib/util/fs/index.ts

```diff
@@ -79,3 +79,12 @@ export async function ensureCacheDir(
   await fs.ensureDir(cacheDirName);
   return cacheDirName;
 }
+
+export function localPathExists(pathName: string): Promise<boolean> {
+  // Works for both files as well as directories
+  return new Promise<boolean>((resolve) => {
+    fs.stat(join(localDir, pathName), (err) => {
+      resolve(!err);
+    });
+  });
```

Comment:
Why not use await with try catch?

Reply:
Wasn't sure if this function supported it

Reply:
we are using it in gradle-wrapper

https://github.com/renovatebot/renovate/blob/5d99ff9862bac803dcea4e6ded5f3fec34abccfc/lib%2Fmanager%2Fgradle-wrapper%2Fartifacts.ts#L63